From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: ArtisanROM <artisanrom@localhost>
Date: Sun, 19 Oct 2026 00:00:00 +0000
Subject: [PATCH] Disable MOBILEAP_11AX support (S9280 BP4A layout)

The S9280 DZI1 semwifi-service.jar restructured showSecProductFeature()
and made SemWifiApChipSupportInfo.isSupportedSoftAp11ax() public, so the
donor-wide patch no longer applies. Adapt it: compute SPF_11ax from the
chip feature array at runtime and consult it from
supportThisSoftApFeature(1).
---
diff -ruN --no-dereference a/smali/com/samsung/android/server/wifi/ap/SemSoftApConfiguration.smali b/smali/com/samsung/android/server/wifi/ap/SemSoftApConfiguration.smali
--- a/smali/com/samsung/android/server/wifi/ap/SemSoftApConfiguration.smali	2026-10-19 06:56:34.436723958 +0800
+++ b/smali/com/samsung/android/server/wifi/ap/SemSoftApConfiguration.smali	2026-10-19 06:56:44.454689589 +0800
@@ -1996,7 +1996,7 @@
 .end method
 
 .method public static showSecProductFeature()Ljava/lang/String;
-    .locals 4
+    .locals 5
 
     new-instance v0, Ljava/lang/StringBuffer;
 
@@ -2028,7 +2028,44 @@
 
     invoke-virtual {v0, v1}, Ljava/lang/StringBuffer;->append(Ljava/lang/String;)Ljava/lang/StringBuffer;
 
-    const-string v1, "SPF_Enhanced=true\nSPF_Enhanced_Menu=d2d\nSPF_Guest=lite\nSPF_MobileAp=true\nSPF_Concurrency=true\nSPF_11ax=true\nSPF_6G=true\nSPF_Dongle=false\nSPF_DualAp=true\n"
+    const-string v1, "SPF_Enhanced=true\nSPF_Enhanced_Menu=d2d\nSPF_Guest=lite\nSPF_MobileAp=true\nSPF_Concurrency=true\n"
+
+    invoke-virtual {v0, v1}, Ljava/lang/StringBuffer;->append(Ljava/lang/String;)Ljava/lang/StringBuffer;
+
+    new-instance v1, Ljava/lang/StringBuilder;
+
+    const-string v3, "SPF_11ax="
+
+    invoke-direct {v1, v3}, Ljava/lang/StringBuilder;-><init>(Ljava/lang/String;)V
+
+    invoke-static {}, Lcom/samsung/android/server/wifi/SemWifiInjector;->getInstance()Lcom/samsung/android/server/wifi/SemWifiInjector;
+
+    move-result-object v3
+
+    invoke-virtual {v3}, Lcom/samsung/android/server/wifi/SemWifiInjector;->getWifiApChipInfo()Lcom/samsung/android/server/wifi/ap/SemWifiApChipInfo;
+
+    move-result-object v3
+
+    const/4 v4, 0x1
+
+    invoke-virtual {v3, v4}, Lcom/samsung/android/server/wifi/ap/SemWifiApChipInfo;->supportThisSoftApFeature(I)Z
+
+    move-result v3
+
+    invoke-virtual {v1, v3}, Ljava/lang/StringBuilder;->append(Z)Ljava/lang/StringBuilder;
+
+    invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+
+    invoke-virtual {v1}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
+
+    move-result-object v1
+
+    invoke-virtual {v0, v1}, Ljava/lang/StringBuffer;->append(Ljava/lang/String;)Ljava/lang/StringBuffer;
+
+    const-string v1, "SPF_6G=true\nSPF_Dongle=false\nSPF_DualAp=true\n"
+
+    invoke-virtual {v0, v1}, Ljava/lang/StringBuffer;->append(Ljava/lang/String;)Ljava/lang/StringBuffer;
+
 
     invoke-virtual {v0, v1}, Ljava/lang/StringBuffer;->append(Ljava/lang/String;)Ljava/lang/StringBuffer;
 
diff -ruN --no-dereference a/smali/com/samsung/android/server/wifi/ap/SemWifiApChipInfo.smali b/smali/com/samsung/android/server/wifi/ap/SemWifiApChipInfo.smali
--- a/smali/com/samsung/android/server/wifi/ap/SemWifiApChipInfo.smali	2026-10-19 06:56:34.438117038 +0800
+++ b/smali/com/samsung/android/server/wifi/ap/SemWifiApChipInfo.smali	2026-10-19 06:56:44.454878651 +0800
@@ -1718,24 +1718,37 @@
 
     const/4 v0, 0x1
 
-    if-eq p1, v0, :cond_2
+    const/4 v2, 0x0
 
-    const/4 v1, 0x2
+    if-ne p1, v0, :cond_2
 
-    if-eq p1, v1, :cond_2
+    iget-object p1, p0, Lcom/samsung/android/server/wifi/ap/SemWifiApChipInfo;->mSemWifiApChipSupportInfo:Lcom/samsung/android/server/wifi/ap/SemWifiApChipInfo$SemWifiApChipSupportInfo;
 
-    const/4 v1, 0x3
+    if-eqz p1, :cond_1
 
-    const/4 v2, 0x0
+    invoke-virtual {p1}, Lcom/samsung/android/server/wifi/ap/SemWifiApChipInfo$SemWifiApChipSupportInfo;->isSupportedSoftAp11ax()Z
+
+    move-result p1
 
-    if-eq p1, v1, :cond_0
+    if-eqz p1, :cond_1
 
+    goto :goto_0
+
+    :cond_1
     return v2
 
-    :cond_0
+    :cond_2
+    const/4 v1, 0x2
+
+    if-eq p1, v1, :goto_0
+
+    const/4 v1, 0x3
+
+    if-ne p1, v1, :cond_1
+
     iget-boolean p1, p0, Lcom/samsung/android/server/wifi/ap/SemWifiApChipInfo;->mSupportPowerSave:Z
 
-    if-nez p1, :cond_2
+    if-nez p1, :goto_0
 
     iget-object p0, p0, Lcom/samsung/android/server/wifi/ap/SemWifiApChipInfo;->mSemWifiApChipSupportInfo:Lcom/samsung/android/server/wifi/ap/SemWifiApChipInfo$SemWifiApChipSupportInfo;
 
@@ -1747,12 +1760,6 @@
 
     if-eqz p0, :cond_1
 
-    goto :goto_0
-
-    :cond_1
-    return v2
-
-    :cond_2
     :goto_0
     return v0
 .end method
